ex2_system/source/leop/leop.c

```diff
@@ -24,103 +24,95 @@
+    if ((switchstatus(sw) != deployed_state)) {
+        sys_log(WARN, "LEOP: %s does not report deployed\n", deployable_to_str(sw));
+    } else {
+        sys_log(INFO, "LEOP: %s reports deployed", deployable_to_str(sw));
+    }
+}
+
+/**
+ * @brief
+ *      Deploy all deployable systems
+ * @return void
+ *      No need for a return, LEOP success is determined by hope
+ */
+void deploy_all_deployables() {
+    int dfgm_switch_deployed_state = 1;
+    deploy(DFGM, MAX_RETRIES, dfgm_switch_deployed_state);
     vTaskDelay(TWO_MIN_DELAY);
 
-    for (getStatus_retries = 0; getStatus_retries <= MAX_RETRIES; getStatus_retries++) {
-        // Deploy UHF
-        for (sw = UHF_P; sw <= UHF_N; sw++) {
-            if ((switchstatus(sw) != 1) && (getStatus_retries != MAX_RETRIES)) {
-                ex2_log("Check #%d: %c not deployed\n", &getStatus_retries, sw);
-                ex2_log("Activated %c\n", sw);
-                activate(sw);
-                vTaskDelay(TWENTY_SEC_DELAY);
-            } else if ((switchstatus(sw) != 1) && (getStatus_retries == MAX_RETRIES)) {
-                ex2_log("Check #%d: %c not deployed, exiting the LEOP sequence.\n", &getStatus_retries, sw);
-                return false;
-            }
-        }
+    int uhf_switch_deployed_state = 1;
+    for (Deployable_t sw = UHF_P; sw <= UHF_N; sw++) {
+        deploy(sw, MAX_RETRIES, uhf_switch_deployed_state);
     }
-    ex2_log("UHF Port deployed, burnwire current = %d\n", burnwire_currents[1]);
-    ex2_log("UHF Zenith deployed, burnwire current = %d\n", burnwire_currents[2]);
-    ex2_log("UHF Starboard deployed, burnwire current = %d\n", burnwire_currents[3]);
-    ex2_log("UHF Nadir deployed, burnwire current = %d\n", burnwire_currents[4]);
 
     vTaskDelay(FOUR_MIN_DELAY);
-
-    for (getStatus_retries = 0; getStatus_retries <= MAX_RETRIES; getStatus_retries++) {
-        // Deploy solar panels
-        for (sw = Port; sw <= Starboard; sw++) {
-            if ((switchstatus(sw) != 1) && (getStatus_retries != MAX_RETRIES)) {
-                ex2_log("Check #%d: %c not deployed\n", &getStatus_retries, sw);
-                ex2_log("Activated %c\n", sw);
-                activate(sw);
-                vTaskDelay(TWENTY_SEC_DELAY);
-            } else if ((switchstatus(sw) != 1) && (getStatus_retries == MAX_RETRIES)) {
-                ex2_log("Check #%d: %c not deployed, exiting the LEOP sequence.\n", &getStatus_retries, sw);
-                return false;
-            }
-        }
-        ex2_log("Port Deployable Panel deployed, burnwire current = %d\n", burnwire_currents[5]);
-        ex2_log("Deployable Payload deployed, burnwire current = %d\n", burnwire_currents[6]);
-        ex2_log("Starboard Deployable deployed, burnwire current = %d\n", burnwire_currents[7]);
+#if HYPERION_PANEL_2U == 1
+    int hyperion_switch_deployed_state = 0; // flight hardware switched soldered on backwards :)
```

Comment:
What?

Reply:
Uh oh. The code is written in a way that this will be easy to fix that when you're sure

Reply:
I'm telling you that based on my hardware testing there is no need for this code. Could you please remove it now and I can add it later if necessary? I think the likelihood of it being necessary is ~nil.

Reply:
I already removed this code with https://github.com/AlbertaSat/ex2_obc_software/commit/37f2d93497ad8139c7fed6d01795e29599f26899

The thing about the deployed state of the switch is easy to change, it's just a constant.